src/callback.rs

```diff
@@ -35,6 +35,31 @@ impl PyCallbackOutput for () {
     const ERR_VALUE: Self = ();
 }
 
+/// A helper trait to result-wrap the return values of functions and methods.
```

Comment:
For new code which is only public because it's proc-macro implementation detail, I've been shoving it somewhere underneath `src/impl_`. Think it makes sense to put this new trait in there?